pontoon/projects/views.py

```diff
@@ -81,18 +84,29 @@ def project(request, slug):
 def ajax_teams(request, slug):
     """Project Teams tab."""
     project = get_object_or_404(
-        Project.objects.visible_for(request.user).available(), slug=slug
+        cast(
+            BaseManager[Project], Project.objects.visible_for(request.user).available()
+        ),
+        slug=slug,
     )
 
-    locales = Locale.objects.available()
+    locales = cast(BaseManager[Locale], Locale.objects.available()).order_by("name")
 
     # Only include filtered teams if provided
     teams = request.GET.get("teams", "").split(",")
     filtered_locales = Locale.objects.filter(code__in=teams)
     if filtered_locales.exists():
         locales = locales.filter(pk__in=filtered_locales)
 
-    locales = locales.prefetch_project_locale(project).order_by("name")
+    latest_trans_ids = project.project_locale.values_list(
+        "latest_translation_id", flat=True
+    )
+    latest_activities = {
+        trans.locale_id: trans.latest_activity
+        for trans in Translation.objects.filter(id__in=latest_trans_ids).select_related(
+            "user", "approved_user"
+        )
+    }
```

Comment:
Shouldn't we use the same pattern in `ajax_projects`? We're making 24 DB queries there vs 7 here and I wonder if that's the culprit.